packages/documentsearch/src/searchview.tsx

```diff
@@ -304,30 +310,62 @@ function UpDownButtons(props: IUpDownProps) {
   const prevShortcut = prevKeys ? ` (${prevKeys})` : '';
   const nextShortcut = nextKeys ? ` (${nextKeys})` : '';
 
+  const upButtonTitle = `${props.trans.__('Previous Match')}${prevShortcut}`;
+  const upButton = props.isEnabled ? (
+    <button
+      className={BUTTON_WRAPPER_CLASS}
+      onClick={() => props.onHighlightPrevious()}
+      tabIndex={0}
+      title={upButtonTitle}
+    >
+      <caretUpEmptyThinIcon.react
+        className={classes(UP_DOWN_BUTTON_CLASS, BUTTON_CONTENT_CLASS)}
+        tag="span"
+      />
+    </button>
+  ) : (
+    <button
+      className={DISABLED_BUTTON_WRAPPER_CLASS}
+      title={upButtonTitle}
+      disabled={true}
+    >
+      <caretUpEmptyThinIcon.react
+        className={classes(DISABLED_UP_DOWN_BUTTON_CLASS, BUTTON_CONTENT_CLASS)}
+        tag="span"
+      />
+    </button>
+  );
+
+  const downButtonTitle = `${props.trans.__('Next Match')}${nextShortcut}`;
+  const downButton = props.isEnabled ? (
+    <button
+      className={BUTTON_WRAPPER_CLASS}
+      onClick={() => props.onHighlightNext()}
+      tabIndex={0}
+      title={downButtonTitle}
+    >
+      <caretDownEmptyThinIcon.react
+        className={classes(UP_DOWN_BUTTON_CLASS, BUTTON_CONTENT_CLASS)}
+        tag="span"
+      />
+    </button>
+  ) : (
+    <button
+      className={DISABLED_BUTTON_WRAPPER_CLASS}
+      title={downButtonTitle}
+      disabled={true}
+    >
+      <caretDownEmptyThinIcon.react
+        className={classes(DISABLED_UP_DOWN_BUTTON_CLASS, BUTTON_CONTENT_CLASS)}
+        tag="span"
+      />
+    </button>
```

Comment:
This comes down to a stylistic choice but I would advocate for keeping it DRY and not adding an extra class name but using the `[disabled]` attribute selector with the existing class name - unless there is a strong reason for the implementation you suggest?

```suggestion
  const downButton = <button
      className={BUTTON_WRAPPER_CLASS}
      onClick={() => props.isEnabled ? props.onHighlightNext() : null}
      tabIndex={0}
      disabled={props.isEnabled}
      title={downButtonTitle}
    >
      <caretDownEmptyThinIcon.react
        className={classes(UP_DOWN_BUTTON_CLASS, BUTTON_CONTENT_CLASS)}
        tag="span"
      />
    </button>
```

Reply:
I had tried the `[disabled]` selector on `button` elements, but I found that in cases where the Find box was empty, the rule did not trigger as expected. There are other rules on the respective classes that caused UI elements to look active (e.g., color changes on hover) even when they were disabled and did nothing when clicked. This fixed the functionality problem, but not the appearance problem.

Reply:
It sounds like an issue with the selectors you tried. I can take a look if you would like to give it another go.